src/components/SlideButton/SlideButton.tsx

```diff
@@ -0,0 +1,137 @@
+"use client";
+
+import React, { CSSProperties, ReactNode, useEffect, useRef } from "react";
+import { PanInfo, useAnimate, useMotionValue, useTransform } from "framer-motion";
+
+import * as S from "./SlideButton.style";
+
+interface SlideButtonProps {
+  buttonColor?: string;
+  textColor?: string;
+  subButtonColor?: string;
+  subTextColor?: string;
+  children: ReactNode | string;
+  subChildren: ReactNode | string;
+  width?: string | number;
+  height?: string | number;
+  subButtonWidth?: number;
+  style?: CSSProperties;
+  onButtonClickHandler?: () => void;
+  onSubButtonClickHandler?: () => void;
+  [key: string]: any;
+}
+
+const SlideButton = ({
+  children,
+  subChildren,
+  buttonColor,
+  textColor,
+  onButtonClickHandler = () => null,
+  onSubButtonClickHandler = () => null,
+  subButtonColor,
+  subTextColor,
+  width,
+  height,
+  subButtonWidth = 90,
+  style,
+}: SlideButtonProps) => {
+  const [scope, animate] = useAnimate();
+  const x = useMotionValue(0);
+  const btnOpacity = useTransform(x, [-(subButtonWidth - 10), -15], [1, 0]);
+
+  const dragStarted = useRef(false);
```

Comment:
[ P4 ] 요 친구는 드래그 중 이라는 상태를 의미하는 친구라면
네이밍을  isDragging으로 사용해보시는건 어떨까요! 변경하시면 저희 boolean 값에 대한 컨벤션에도 잘 맞을 것 같습니다! 👍

```suggestion
  const isDragging = useRef(false);
```

Reply:
네 반영하도록 하겠습니다!